File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-0.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        //first easiest to iterate througb if it is a charcater array
        char[] arr= s.toCharArray();
        //maybe same approach to the longest substrign of numbers
        //so we create a set or something to add a charcater until we reach a duplicat ehcarcter, and create a set of charcetr we alreaysaw
        //as well a two pointer approach
        HashSet<Character> seen= new HashSet<>();
        //left pointer start of string, rigth pointer rigth next to next
        //if the character at the pointers are the same
        //move the left pointer and the rigth pointer to the left
        //if it isnt the same, add to the alreadySaw set
        //continue
        int l= 0;
        int maxLength=0;
        for(int r=0;r<arr.length;r++)
        {
            char rightChar= arr[r];
            //removing the duplicate when we fidn one and then moving the window
            while (seen.contains(rightChar)) {
                seen.remove(arr[l]);
                l++;
            }
            if (!seen.contains(rightChar)) {
                seen.add(rightChar);
            }

            maxLength = Math.max(maxLength, r - l + 1);
        }
        
        return maxLength;

    }
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        #i would use a two pointer approach
        # start pointer= 0 
        # end pointer= array.length()-1
        # we compare the  number at the end + first and comapre it to the atgret
        #array[start pointer]+ array[end pointer] 
        #while start pointer < end pointer:
        #    if(start pointer + end pointer == target):
        #       return [start pointer, end pointer]
        #   if(start pointer + end pointer> target):
        #       move the end pointer left
        #   else if(start pointer +end pointer <target):
        #       move the right pointer right

        start_pointer, end_pointer = 0, len(numbers)-1

        while start_pointer < end_pointer:
            current_sum = numbers[start_pointer] + numbers[end_pointer]
            if(current_sum ==target):
                return [start_pointer + 1, end_pointer + 1] 
            elif (current_sum< target):
                 start_pointer +=1
            else:
                end_pointer -=1
        


